3DAmsterdam/Assets/Netherlands3D/Scripts/Traffic/Car.cs

```diff
@@ -75,21 +142,64 @@ void Update()
                             currentRoadIndex++;
                         }
 
-                        Vector3 temp = transform.position;
-                        temp.y = 50f;
+                        Vector3 carPos = transform.position;
 
-                        if (Physics.Raycast(temp, -Vector3.up, out hit, Mathf.Infinity))
+
+                        //ask wheels where their position is
+                        //calculate center point
+                        //rotate car body based on center point
+                        carPos.y += 1.5f;
+                        if (Physics.Raycast(carPos, transform.forward, out hit, 20f))
                         {
                             // if the map tiles are loaded beneath the car
-                            MoveCar(hit.point);
+                            if (hit.collider.gameObject.name == "BasicTruck" || hit.collider.gameObject.name == "BasicCar")
```

Comment:
here you might want to use tags. so create a shared tag that all vehicles use, so you only have to do one comparison.
for example: if(hit.collider.CompareTag("Vehicle"))

the CompareTag method should also be faster than a direct string comparison

Reply:
I had this written as a TO:DO and never got around to it, I fixed this by adding a Traffic layer and comparing that